lib/rules/indent.js

```diff
@@ -0,0 +1,378 @@
+/*eslint no-use-before-define:[2, "nofunc"]*/
+/**
+ * @fileoverview This option sets a specific tab width for your code
+ * @author Dmitriy Shekhovtsov
+ * @copyright 2014 Dmitriy Shekhovtsov. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var indentChar = " ";
+    var indentSize = 4;
+    var lines = null;
+    var indentStack = [0];
+    var linesToCheck = null;
+
+    if (context.options.length) {
+        if (context.options[0] === "tab") {
+            indentChar = "\t";
+            indentSize = 1;
+        } else if (typeof context.options[0] === "number") {
+            indentSize = context.options[0];
+        }
+    }
+
+    // variables should be defined here
+
+    var blockParents = [
+        "IfStatement",
+        "WhileStatement",
+        "DoWhileStatement",
+        "ForStatement",
+        "ForInStatement",
+        "ForOfStatement",
+        "FunctionDeclaration",
+        "FunctionExpression",
+        "ArrowExpression",
+        "CatchClause"
```

Comment:
@lo1tuma what is `with`? :) I hope it will be deprecated some day,
got your point anyway